event-store/impl-fs/src/commonMain/kotlin/dev/eskt/store/impl/fs/FileSystemStorage.kt

```diff
@@ -137,6 +137,17 @@ public class FileSystemStorage internal constructor(
         }
     }
 
+    public override fun <E, I, R> useStreamEvents(
+        streamType: StreamType<E, I>,
+        streamId: I,
+        sinceVersion: Int,
+        consume: (Sequence<EventEnvelope<E, I>>) -> R,
+    ): R {
+        // TODO optimize this implementation, we don't need to load the entire list
+        val events = getStreamEvents(streamType, streamId, sinceVersion)
+        return consume.invoke(events.asSequence())
```

Comment:
So we're not getting any of the benefits of this yet?

Reply:
the file system storage is just a proof-of-concept, we are using the PostgreSQL storage currently.